app/utils/profileManagement.js

```diff
@@ -0,0 +1,66 @@
+import { DISPLAY_NAME_FIELD } from 'containers/Profile/constants';
+
+import { saveText, getText, getImage } from './ipfs';
+import { getTableRow, sendTransaction } from './eosio';
+
+export async function uploadImageFile(image) {
+  const savedText = await saveText(image);
+  return savedText;
+}
+
+export async function getImageFile(imageHash) {
+  const image = await getImage(imageHash);
+  return image;
+}
+
+export async function getBlob(canvas) {
+  const res = await fetch(canvas);
+  const blob = await res.blob();
+  return blob;
+}
+
+export async function getProfileInfo(profileHash) {
+  const eos = await getTableRow('account', 'allaccounts', profileHash);
+  const ipfs = await getText(eos.ipfs_profile);
+  const ipfsParsed = JSON.parse(ipfs);
+  const savedProfileImg = ipfsParsed.savedProfileImg
+    ? await getImageFile(ipfsParsed.savedProfileImg)
+    : '';
+
+  return {
+    eos,
+    ipfs: ipfsParsed,
+    savedProfileImg,
+  };
+}
+
+export async function saveProfile(owner, profile) {
+  const ipfsProfile = await saveText(JSON.stringify(profile));
+
+  const setipfspro = new Promise(async res => {
+    await sendTransaction(owner, 'setipfspro', {
+      owner,
+      ipfs_profile: ipfsProfile,
+    });
+    res();
+  });
+
+  const setdispname = new Promise(async res => {
+    await sendTransaction(owner, 'setdispname', {
+      owner,
+      display_name: profile[DISPLAY_NAME_FIELD] || '',
+    });
+    res();
+  });
+
+  return Promise.all([setipfspro, setdispname]);
```

Comment:
Promise.all can be used with async functions. No need to wrap into promise

https://www.dalejefferson.com/async-await-promise-all-array-destructuring/